hermes/probe/mock/mock_gcs.go

```diff
@@ -0,0 +1,88 @@
+package mock
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+)
+
+type fakeBucket struct {
+	attrs   *storage.BucketAttrs
+	objects map[string][]byte
+}
+
+type fakeClient struct {
+	stiface.Client
+	buckets map[string]*fakeBucket
+}
+
+type fakeBucketHandle struct {
+	stiface.BucketHandle
+	c    *fakeClient
+	name string
+}
+
+type fakeObjectHandle struct {
+	stiface.ObjectHandle
+	c          *fakeClient
+	bucketName string
+	name       string
+}
+
+type fakeWriter struct {
+	stiface.Writer
+	obj fakeObjectHandle
+	buf bytes.Buffer
+}
+
+func NewFakeClient() stiface.Client {
+	return &fakeClient{buckets: map[string]*fakeBucket{}}
+}
+
+func (o fakeObjectHandle) NewWriter(context.Context) stiface.Writer {
```

Comment:
Recommend using pointers for receivers in general.
Although this method doesn't modify the receiver, I would still use a pointer, as it's cheaper.